### DISCUSSION TITLE
Pretraining while keeping some layers with Dropout frozen

### DISCUSSION THREAD

**Discussion by serchsm** [speechbrain/speechbrain#1810] [Q&A]
Hello,

 I am trying to fine tune only a few layers of a model while keeping the base layers frozen. The frozen layers have `Dropout` so I need them frozen in `eval` mode. According to [Pretrain and Fine-Tune](https://colab.research.google.com/drive/1LN7R3U3xneDgDRK2gC5MzGkLysCWxuC3?usp=sharing) the workflow is to load the pre-trained components using `Pretrainer` (takes care of freezing and eval mode), and then leverage the `fit` method in the `Brain` class to only train some of the modules. 

The problem that I found is that the `fit` method calls `_fit_train` which calls `self.modules.train()` [here](https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/core.py#L1025) at the begining of the method. So, the `pretrained` modules with `Dropout` end up behaving differently (back to `train`).

In the `yaml`, I define the `modules` like this:

```
modules:
  encoder: !ref <encoder>
  predictor_base: !ref <predictor_base>
  predictor_linear: !ref <predictor_linear>
  Tjoint: !ref <Tjoint>
  transducer_linear: !ref <transducer_linear>
```
 The `encoder` and `predictor_base` are `Pretrained` modules (loaded using `Pretrainer`). Since I want them **frozen** and they contain `**Dropout**` layers, it seems I cannot include them in the `modules` section of the `yaml`, beacuse they eventually are going to be set back to `train`.  Am I reading this correctly for my use case? 
Is the solution here to exclude the pretrained components (`encoder` and `predictor_base`) from modules? Then in `compute_forward` I can still access them by calling `self.hparams.encoder` and `self.hparams.predictor_base`. It seems this is a particular case because they contain `Dropout` layers which behave differently (train vs eval), tho I imagine that a lot of people have pre-trained modules with this type of layers.
